### DISCUSSION TITLE
Add Docker support for reproduction repos

### DISCUSSION THREAD

**Discussion by yandodov** [payloadcms/payload#2493] [Feature Requests & Ideas]
I want to report a bug and I'm following the [reproduction guide](https://github.com/payloadcms/payload/blob/master/.github/reproduction-guide.md). However, I'm using MongoDB Atlas for my regular projects and really have no need for a database locally. Therefore, I don't have MongoDB on my machine at all. Since it's a company-provided machine, installing new software isn't that straightforward, due to security policies.

I'd be awesome if Payload also provided a ready-to-go docker image that you could just build and get up and running instantly. Perhaps the dev server could be baked into it as well, so you don't even need to manage your Node.js versions.

This would greatly reduce the friction with reporting (and maybe even fixing) issues, leading to more contributions and faster iterations of the product.

**Comment by yandodov** [comment]
For people out there with the same struggle, you can just run MongoDB from Docker as a standalone image and map port 27017 of your host machine to it.

[It's explained in the MongoDB docs.](https://www.mongodb.com/docs/manual/tutorial/install-mongodb-community-with-docker/#procedure) The only difference is that in their guide, the run command doesn't map the ports (doesn't specify the `-p` flag):

```shell
docker pull mongodb/mongodb-community-server
docker run --name mongo -p 27017:27017 -d mongodb/mongodb-community-server:latest
```

After this, you'd get MongoDB running in the background and you can connect to it over `http://localhost:27017` via MongoDB Compass or simply run `yarn dev _community` to start Payload.

If you don't expose the port (`-p 27017:27017`), you'd get the following error upon trying to connect:

> connect ECONNREFUSED 127.0.0.1:27017

---

Still, it'd be ideal if Payload provides a ready-made container with the proper MongoDB versions that it supports and whatnot.

**Comment by teunmooij** [comment]
If you set node_env to `test` Payload automatically switches to [mongodb-memory-server](https://www.npmjs.com/package/mongodb-memory-server). You can add that package as a dev dependency. Of course you will need to add some seed functions if you want data.
